### ISSUE TITLE
Query a node which is an array itself

### ISSUE THREAD

**Issue by joyceview** [angular/angularfire2#403]
Hi there,

Is there a way to query a node which is an array? What I want is return this node of "-KO8cSA1a6Tl21vHJao5" when today is Monday or Tuesday.

"data" : { 
  "-KO8cSA1a6Tl21vHJao5" : {
    "repeatDays" : [ "Mon", "Tue" ],
    "updatedOn" : "Mon Aug 01 2016 22:06:12 GMT-0700 (PDT)"
  }
}

I also tried to save the value as a string instead of string[], but can't get the result either.

Could anyone help me with this?

Thanks a lot!

**Comment by lexcaraig** [comment]
You mean, you want to get the nodes that has a repeatDays of `Mon && Tue` ?

**Comment by joyceview** [comment]
yes

**Comment by joyceview** [comment]
Yes, If today is Mon or Tue, I want to get the node.

**Comment by lexcaraig** [comment]
You can refer to this issue #396

**Comment by joyceview** [comment]
I'm using the filter. 
The filter is working only if the value is string, not working if the value is an array or a string of "Mon,Tue". 
As my today's variable could only be "Mon" or "Tue", but my RepeatDays in database could be up to 7 days.

If there is a comparison of Contains or something alike, that might work.

**Comment by joyceview** [comment]
I'm using the filter. 
The filter is working only if the value in firebase is a string. But not working if the value is an array, nor as a string like "Mon,Tue". 
As my today's variable could only be "Mon" or "Tue", but my RepeatDays in database could be up to 7 days.

If there is a comparison of Contains or something alike, that might work.

**Comment by r-park** [comment]
This should work if you store `repeatDays` as a string `'Mon,Tue'`:
```js
import { ReplaySubject } from 'rxjs/ReplaySubject';

const equalTo$ = new ReplaySubject(1);

const query$ = af.database.list('/data', {query: {
  orderByChild: 'repeatDays',
  equalTo: equalTo$
}});

equalTo$.next('Mon,Tue');

query$.subscribe(data => console.log(data)); //=> [{repeatDays : 'Mon,Tue' ... }]
```

**Comment by joyceview** [comment]
Thank you r-park for your reply. Sorry that I didn't really get it.

I've updated the data saved in Firebase as 'Mon,Tue'. And also added { ReplaySubject }. The 'Mon,Tue' is the data saved in database, I don't know how to set it to equalTo$.

Here is my code, would you mind to take a look and add the logic there? I tried but didn't work. The data is under the uid node, I don't know how to add the logic.

getUserTodayPrayers() {
    var prayers: FirebaseListObservable<any>;
    var todayDayName = (new Date()).toDateString().split(" ")[0];

    this.af.auth.subscribe(auth => {
      prayers = this.af.database.list(`users/${auth.uid}/prayers`,{
          query: {
            orderByChild: 'repeatDays',
            equalTo: todayDayName
          }
        }
      );
    });
    
    return prayers;
  }


Thanks a lot!

**Comment by joyceview** [comment]
Sorry to bother again, but the issue is still not solved. I did try the code above, but it's not working.

The scenario is this:
KeyValue pair saved in database is this: { repeatDays:  'Mon, Tue' }
And today's day name is:  'Mon'

The result I'm expecting is, I'll be able to get the data node with the key value pair as above. Today's day name is only partial of the data saved in database.

The answer I got above can only get me the data node when the value are exactly same. I need to get the value when it's partially there.
